File: src/main/java/fi/aalto/cs/apluscourses/ui/courseproject/CourseProjectActionDialogs.java
package fi.aalto.cs.apluscourses.ui.courseproject;

import com.intellij.openapi.project.Project;
import fi.aalto.cs.apluscourses.presentation.CourseProjectViewModel;
import fi.aalto.cs.apluscourses.presentation.CourseSelectionViewModel;
import org.jetbrains.annotations.NotNull;

public interface CourseProjectActionDialogs {

  boolean showCourseSelectionDialog(@NotNull Project project,
                                    @NotNull CourseSelectionViewModel courseSelectionViewModel);

  boolean showMainDialog(@NotNull Project project,
                         @NotNull CourseProjectViewModel courseProjectViewModel);

  boolean showRestartDialog();

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/ui/temp/courseproject/CourseProjectActionDialogs.kt
package fi.aalto.cs.apluscourses.ui.temp.courseproject

import com.intellij.openapi.ui.Messages
import fi.aalto.cs.apluscourses.MyBundle.message

class CourseProjectActionDialogs {
    fun showRestartDialog(): Boolean {
        return Messages.showOkCancelDialog(
            message("ui.courseProject.dialogs.showRestartDialog.message"),
            message("ui.courseProject.dialogs.showRestartDialog.title"),
            message("ui.courseProject.dialogs.showRestartDialog.okText"),
            message("ui.courseProject.dialogs.showRestartDialog.cancelText"),
            Messages.getQuestionIcon()
        ) == Messages.OK
    }
}
